apollo-router/src/plugins/reporting/otlp/mod.rs

```diff
@@ -98,15 +99,31 @@ impl Tracing {
     }
 }
 
-#[derive(Debug, Clone, Deserialize, Serialize, Default)]
+#[derive(Debug, Clone, Deserialize, Serialize, Default, JsonSchema)]
 #[serde(deny_unknown_fields, rename_all = "snake_case")]
 pub struct ExportConfig {
     #[serde(deserialize_with = "endpoint_url", default)]
     pub endpoint: Option<Url>,
+
+    #[schemars(schema_with = "option_protocol_schema")]
     pub protocol: Option<Protocol>,
     pub timeout: Option<u64>,
 }
 
+fn option_protocol_schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+    Option::<ProtocolMirror>::json_schema(gen)
+}
+
+#[derive(JsonSchema)]
+#[allow(dead_code)]
+enum ProtocolMirror {
+    /// GRPC protocol
+    Grpc,
+    // HttpJson,
+    /// HTTP protocol with binary protobuf
+    HttpBinary,
+}
+
```

Comment:
NIT: I presume this is a restricted copy of the telemetry Protocol struct which only contains the protocols we support.

In which case, maybe rename to `ImplementedProtocol` or `RestrictedProtocol` or `ProtocolSubset`...?

(It's definitely not a Mirror and that name doesn't give much of a hint about its purpose.)

Reply:
It's not even that, it is an exact copy of the otel struct. I'll add a comment though.